src/string.js

```diff
@@ -172,3 +172,26 @@ export function trimStart(str) {
 
   return (str + '_').trim().slice(0, -1);
 }
+
+/**
+ * Wrapper around String.replace that handles asynchronous resolution.
+ * @param {string} str
+ * @param {RegExp} regex
+ * @param {Function} replacer
+ * @return {!Promise<string>}
+ */
+export function asyncStringReplace(str, regex, replacer) {
```

Comment:
Please add unit test for this function

Reply:
Done.

Reply:
Actually hit a bug here with variable number of capture groups. Didn't think of this when we moved to make it generic. Working on fix.

Reply:
OK fixed :) should work with any number of capture groups.